### PR TITLE
docs: revamp example docs

### PR DESCRIPTION
**Pull request by rithikJha**
Before: 
Screenshot 2024-08-07 015834

After:
Screenshot 2024-08-07 015852
Screenshot 2024-08-07 015934
Screenshot 2024-08-07 015941
Screenshot 2024-08-07 015949

### CODE REVIEW

File: docs/src/examples/python_examples/recommender_systems.md
Comment (AyushExel) [inline docs/src/examples/python_examples/recommender_systems.md:1]: This file seems empty? Is it intentional? can we add it later once it has some content?

File: docs/src/examples/python_examples/ai_agents.md
Comment (AyushExel) [inline docs/src/examples/python_examples/ai_agents.md:1]: This also

**Comment by AyushExel** [comment]
Looks good just a couple of comments.

File: docs/src/examples/python_examples/recommender_systems.md
Comment (rithikJha) [reply docs/src/examples/python_examples/recommender_systems.md:1]: yes, some files are empty ! Okay, will add it later after adding content

**Review by AyushExel — APPROVED** [review]

**Pull request merged**